.github/workflows/beam_PreCommit_Python_ML.yml

```diff
@@ -0,0 +1,112 @@
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+name: PreCommit Python ML tests with ML deps installed
+on:
+  pull_request_target:
+    branches: [ "master", "release-*" ]
+    paths: [ "model/**","sdks/python/**","release/**", 'release/trigger_all_tests.json', '.github/trigger_files/beam_PreCommit_Python_ML.json']
```

Comment:
No-op right now, but we may eventually consider making this only trigger on ml changes

Reply:
Also seen anecdotal evidence of using pytest-testmon, https://stackoverflow.com/questions/67398655/run-only-tests-which-depend-on-the-change.
we could give it an have a full postcommit coverage as backup